Client/Frontend/Browser/HashchangeHelper.swift

```diff
@@ -0,0 +1,33 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import WebKit
+
+protocol HashchangeHelperDelegate: class {
+    func hashchangeHelperDidHashchange(hashchangeHelper: HashchangeHelper)
+}
+
+class HashchangeHelper: NSObject, BrowserHelper, UIGestureRecognizerDelegate {
```

Comment:
As mentioned above, a class-level comment explaining this helper might be useful here. There are two ways to create "documentation comments" in Xcode:
```
/**
 * Style 1.
 */
```
or:
```
/// Style 2.
```
(By documentation comments, I mean the description that pops up in Xcode when you option-click a variable or class name).